### PR TITLE
CV UI - E2E Testing for regular and Composite CVs 

### PR DESCRIPTION
**Pull request by sambible**
Adds 2 tests, converted from the old CV UI tests, and updated for modern behavior. E2E testing for Composite and Component CV workflows. 

Requires: https://github.com/SatelliteQE/airgun/pull/939

### CODE REVIEW

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contentview.py::test_positive_end_to_end
tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e
airgun: 939

File: tests/foreman/ui/test_contentview.py [hunk 1]
```diff
@@ -17,6 +17,71 @@
 
 from robottelo.constants import FAKE_FILE_NEW_NAME, REPOS, DataFile
 
+VERSION = 'Version 1.0'
+
+
+@pytest.mark.tier2
```
Comment (vsedmik) [inline tests/foreman/ui/test_contentview.py:23]:
```suggestion
@pytest.mark.e2e
@pytest.mark.tier2
```

File: tests/foreman/ui/test_contentview.py [hunk 2]
```diff
@@ -17,6 +17,71 @@
...
+    with module_target_sat.ui_session() as session:
+        session.organization.select(org_name=module_org.name)
+        session.contentview_new.create({'name': cv_name})
+        session.contentview_new.add_content(cv_name, repo_name)
+        result = session.contentview_new.publish(cv_name, promote=True, lce=module_lce.name)
+        assert result[0]['Version'] == VERSION
+        assert module_lce.name in result[0]['Environments']
+
+
+@pytest.mark.tier2
```
Comment (vsedmik) [inline tests/foreman/ui/test_contentview.py:53]:
```suggestion
@pytest.mark.e2e
@pytest.mark.tier2
```

File: tests/foreman/ui/test_contentview.py [hunk 3]
```diff
@@ -17,6 +17,71 @@
 
 from robottelo.constants import FAKE_FILE_NEW_NAME, REPOS, DataFile
 
+VERSION = 'Version 1.0'
+
+
+@pytest.mark.tier2
+@pytest.mark.upgrade
+def test_positive_end_to_end(session, module_target_sat, module_org, module_lce, target_sat):
```
Comment (vsedmik) [inline tests/foreman/ui/test_contentview.py:25]:
```suggestion
def test_positive_end_to_end(session, module_target_sat, module_org, module_lce):
```
Common, don't be greedy, use just one of them here and bellow.

File: tests/foreman/ui/test_contentview.py [hunk 4]
```diff
@@ -17,6 +17,71 @@
...
+
+    :steps:
+        1. Create Product/repo and Sync it
+        2. Create CV and add created repo in step1
+        3. Publish and promote it to new LCE
+
+    :expectedresults: content view is created, updated with repo publish and
+        promoted to  the LCE
+
+    :CaseImportance: High
```
Comment (vsedmik) [inline tests/foreman/ui/test_contentview.py:38]:
```suggestion
```
We can omit the importance here and "inherit" the module-wide from L11. Or set it to `Critical`. :thinking:

File: tests/foreman/ui/test_contentview.py [hunk 5]
```diff
@@ -17,6 +17,71 @@
...
+        session.organization.select(org_name=module_org.name)
+        session.contentview_new.create({'name': cv_name})
+        session.contentview_new.add_content(cv_name, repo_name)
+        result = session.contentview_new.publish(cv_name, promote=True, lce=module_lce.name)
+        assert result[0]['Version'] == VERSION
+        assert module_lce.name in result[0]['Environments']
+
+
+@pytest.mark.tier2
+def test_positive_ccv_e2e(session, module_target_sat, module_org, module_lce, target_sat):
```
Comment (vsedmik) [inline tests/foreman/ui/test_contentview.py:54]:
```suggestion
def test_positive_ccv_e2e(session, module_target_sat, module_org, module_lce):
```
Same as above. Or do I actually overlook something? :thinking:

File: tests/foreman/ui/test_contentview.py [hunk 6]
```diff
@@ -17,6 +17,71 @@
...
+    """Create several CVs, and a CCV. Associate some content with each, and then associate the CVs
+    with the CV - everything should work properly.
```
Comment (vsedmik) [inline tests/foreman/ui/test_contentview.py:56]:
```suggestion
    with the CCV - everything should work properly.
```

File: tests/foreman/ui/test_contentview.py [hunk 7]
```diff
@@ -17,6 +17,71 @@
...
+
+    :id: d2db760d-4441-4e6e-9f7d-84501d8e0a13
+
+    :expectedresults: CCV can be created, and CVs can be added to it. Repository count is appropriate.
+
+    :CaseImportance: High
```
Comment (vsedmik) [inline tests/foreman/ui/test_contentview.py:62]: Same as above.

File: tests/foreman/ui/test_contentview.py [hunk 8]
```diff
@@ -17,6 +17,71 @@
...
+            cv_name = module_target_sat.api.ContentView(organization=module_org).create().name
+            session.contentview_new.add_content(cv_name, repo_name)
+            session.contentview_new.publish(cv_name, promote=True, lce=module_lce.name)
+            result = session.contentview_new.add_cv(ccv_name, cv_name)
+            assert result[0]["Status"] == "Added"
+        session.contentview_new.publish(ccv_name)
+        # Check that composite cv has three repositories in the table as we
+        # were using one repository for each content view
+        result = session.contentview_new.read_version_table(ccv_name, VERSION, "repositories")
+        assert len(result) == 1
```
Comment (vsedmik) [inline tests/foreman/ui/test_contentview.py:83]: There is some inconsistency between L80 and L83. Should it be `== 3` or should we extend the assertion?

**Review by vsedmik — CHANGES_REQUESTED** [review]
A few proposals and questions pending in my mind.

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contentview.py::test_positive_end_to_end
tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e
airgun: 939

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contentview.py::test_positive_end_to_end
pytest: tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e
airgun: 939

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8512
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e --external-logging
Test Result : ================== 1 failed, 11 warnings in 723.12s (0:12:03) ==================
```

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e
airgun: 939

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8514
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e --external-logging
Test Result : ================== 1 failed, 11 warnings in 712.50s (0:11:52) ==================
```

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e
airgun: 939

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8515
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e --external-logging
Test Result : ================== 1 failed, 11 warnings in 742.75s (0:12:22) ==================
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8516
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e --external-logging
Test Result : ================== 1 failed, 11 warnings in 702.64s (0:11:42) ==================
```

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e
airgun: 939

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8518
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e --external-logging
Test Result : ================== 1 failed, 11 warnings in 697.94s (0:11:37) ==================
```

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e
airgun: 939

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8538
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e --external-logging
Test Result : ================== 1 failed, 11 warnings in 747.05s (0:12:27) ==================
```

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e
airgun: 939

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e
airgun: 939

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8540
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_contentview.py::test_positive_ccv_e2e --external-logging
Test Result : ================== 1 failed, 11 warnings in 740.08s (0:12:20) ==================
```

File: tests/foreman/ui/test_contentview.py [hunk 9]
```diff
@@ -17,6 +17,77 @@
...
+    repo_name = gen_string('alpha')
+    # Create a product and sync'ed repository
+    module_target_sat.api_factory.create_sync_custom_repo(module_org.id, repo_name=repo_name)
+    with module_target_sat.ui_session() as session:
+        # Creates a composite CV
+        session.organization.select(org_name=module_org.name)
+        session.contentview_new.create({'name': ccv_name}, composite=True)
+        # Create three content-views and add synced repo to them
+        for _ in range(3):
+            cv_name = module_target_sat.api.ContentView(organization=module_org).create().name
```
Comment (vijaysawant) [inline tests/foreman/ui/test_contentview.py:80]: Why this api call to create CV? (to save time in navigation?) 
Can we use `session.contentview_new.create()`?

File: tests/foreman/ui/test_contentview.py [hunk 10]
```diff
@@ -17,6 +17,77 @@
...
+            session.contentview_new.add_content(cv_name, repo_name)
+            session.contentview_new.publish(cv_name, promote=True, lce=module_lce.name)
+            result = session.contentview_new.add_cv(ccv_name, cv_name)
+            assert result[0]["Status"] == "Added"
```
Comment (vijaysawant) [inline tests/foreman/ui/test_contentview.py:84]:
```suggestion
            assert result[0]['Status'] == 'Added'
```

File: tests/foreman/ui/test_contentview.py [hunk 11]
```diff
@@ -17,6 +17,77 @@
...
+        session.contentview_new.publish(ccv_name)
+        # Check that composite cv has one repository in the table as we
+        # were using the same repository for each Content View.
+        result = session.contentview_new.read_version_table(ccv_name, VERSION, "repositories")
```
Comment (vijaysawant) [inline tests/foreman/ui/test_contentview.py:88]:
```suggestion
        result = session.contentview_new.read_version_table(ccv_name, VERSION, 'repositories')
```

**Review by vijaysawant — CHANGES_REQUESTED** [review]

File: tests/foreman/ui/test_contentview.py (hunk 9 above)
Comment (sambible) [reply tests/foreman/ui/test_contentview.py:80]: Saves a ton of time

Comment (LadislavVasina1) [reply tests/foreman/ui/test_contentview.py:80]: @vijaysawant @sambible If there wouldn't be an e2e test that tests the CV creation I would also prefer creating the CVs in this test, but as it is covered in `test_positive_ccv_e2e` I agree with Sam on saving time here and keeping the CV creation through api.

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_contentview.py::test_positive_end_to_end
airgun: 939

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8993
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_contentview.py::test_positive_end_to_end --external-logging
Test Result : ================= 1 failed, 15 warnings in 1204.10s (0:20:04) ==================
```

**Review by LadislavVasina1 — DISMISSED** [review]

**Comment by LadislavVasina1** [comment]
@sambible Could you please address vijays and vladimir's comments that are marked as resolved, but I don't see them in the committed files, thanks.

**Review by LadislavVasina1 — APPROVED** [review]

**Review by ColeHiggins2 — APPROVED** [review]

**Review by vsedmik — APPROVED** [review]

**Review by chris1984 — APPROVED** [review]

**Pull request merged**